Sorochinsky/Bulk_Thumbnail_Creator/Bulk_Thumbnail_Creator/Program.cs

```diff
@@ -0,0 +1,170 @@
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Configuration;
+
+namespace Bulk_Thumbnail_Creator
+    {
+    public class Program
+    {
+        static void Main()
+        {
+            string filesPath = ConfigurationManager.AppSettings["filePath"];
+            string newFilesPath = ConfigurationManager.AppSettings["newFilePath"];
+
+            if (!IsDirectoryExists(filesPath)) return;
+            DirectoryInfo directoryInfo = new DirectoryInfo(filesPath);
+
+            IsDirectoryExists(newFilesPath);
+
+            FileInfo[] files = directoryInfo.GetFiles();
+            if (!IsFileExists(files)) return;  
+
+            CancellationTokenSource cancelTokenSource = new CancellationTokenSource();
+            CancellationToken token = cancelTokenSource.Token;
+
+            do
+            {
+                ShowMenu();
+                MenuPoints choice = InputMenu();
+                Console.Clear();
+
+                if (choice == MenuPoints.Exit) break;
+
+                Task task;
+                bool isCanceled = false;
+
+                switch (choice)
+                {
+                    case MenuPoints.Resize:
+                        {
+                            int width = GetWidth();
+                            int heigth = GetHeigth();
+
+                            task = new Task(() =>
+                            { 
+                                PhotoManager.ResizeImages(filesPath, newFilesPath, width, heigth, token, out isCanceled);
+                            });
+
+                            task.Start();
+                            CancelOperation(cancelTokenSource);
+
+                            break;
+                        }
+                    case MenuPoints.Rename:
+                        {
+                            string newFilesName = GetNewFilesName();
+
+                            task = new Task(() =>
+                            {
+                                PhotoManager.RenameImages(files, newFilesName, token, out isCanceled);
+                            });
+
+                            task.Start();
+                            CancelOperation(cancelTokenSource);
+
+                            break;
+                        }
+                    default:
+                        {
+                            Console.WriteLine(Constants.unknownCommand);
+                            break;
+                        }
+                }
+
+                if (!isCanceled) Console.WriteLine(Constants.successfullOperation);
+                else Console.WriteLine(Constants.canceledOperation);
+                
+                Console.WriteLine(Constants.anyKeyToContinue);
+                Console.ReadKey();
+                Console.Clear();
+
+            } while (true);
+
+            Console.WriteLine(Constants.programEnd);
+            Console.ReadKey();
+        }
+
+        static bool IsDirectoryExists(string filesPath)
```

Comment:
logic for working with directories and communication with user (class Console) - should be splitted up